airflow/providers/google/cloud/hooks/new_dataproc.py

```diff
@@ -0,0 +1,892 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""This module contains a Google Cloud Dataproc hook."""
+
+import time
+import uuid
+import warnings
+from typing import Any, Dict, Iterable, List, Optional, Sequence, Tuple, Union
+
+from cached_property import cached_property
+from google.api_core.exceptions import ServerError
+from google.api_core.retry import Retry
+from google.cloud.dataproc import (  # pylint: disable=no-name-in-module
+    Cluster,
+    ClusterControllerClient,
+    Job,
+    JobStatus,
+    WorkflowTemplate,
+)
+from google.cloud.dataproc_v1beta2 import (  # pylint: disable=no-name-in-module; ClusterControllerClient,
+    JobControllerClient,
+    WorkflowTemplateServiceClient,
+)
+from google.protobuf.internal.well_known_types import FieldMask
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+from airflow.version import version as airflow_version
+
+
+class DataProcJobBuilder:
```

Comment:
Do we need job builder? It's only required for deprecated operators

Reply:
I do not know, I can delete it.